src/app/core/components/header/header-nav/header-nav.component.ts

```diff
@@ -8,13 +9,17 @@ import { Component, OnInit } from '@angular/core';
 export class HeaderNavComponent implements OnInit {
   isSearchOpen = false;
   showMenu: Boolean = false;
+  count = [];
   onSearchBtnClick() {
     this.isSearchOpen = !this.isSearchOpen;
   }
 
-  constructor() { }
+  constructor(
+    private cartService: CartService
+  ) { }
 
   ngOnInit() {
+    this.cartService.getCartLength().subscribe(data => this.count = data);
```

Comment:
You could use only length of the data for `count` here:
```
data => this.count = data.length
```
Also name `getCartLength` suggests you will receive a number, but instead you receive an array of data